napari/_qt/qt_color_dialog.py

```diff
@@ -0,0 +1,267 @@
+import re
+from typing import Optional
+
+import numpy as np
+from qtpy.QtCore import Qt, Signal, Slot
+from qtpy.QtGui import QColor
+from qtpy.QtWidgets import (
+    QColorDialog,
+    QCompleter,
+    QFrame,
+    QHBoxLayout,
+    QLineEdit,
+    QVBoxLayout,
+    QWidget,
+)
+
+from ..layers.utils.color_transformations import ColorType
+from ..utils.colormaps.standardize_color import (
+    hex_to_name,
+    rgb_to_hex,
+    transform_color,
+)
+from .qt_modal import QtPopup
+from vispy.color import get_color_dict
+
+# matches any 3- or 4-tuple of int or float, with or without parens
+# captures the numbers into groups.
+# this is used to allow users to enter colors as e.g.: "(1, 0.7, 0)"
+rgba_regex = re.compile(
+    r"\(?([\d.]+),\s*([\d.]+),\s*([\d.]+),?\s*([\d.]+)?\)?"
+)
+
+TRANSPARENT = np.array([0, 0, 0, 0], np.float32)
+UNKNOWN = -1  # a constant to indicate that the user entered an invalid string
+FALLBACK = TRANSPARENT  # what will be shown if a user entry is UNKNOWN
```

Comment:
Ah looking at the code I see why what I mentioned in the main thread happened, but I did find it unexpected as a user

Reply:
no problem.  we can either make it black, or go back to what it was before.

Reply:
i'd probably have it go back to what it was before, seems like the least confusing. might be worth trying what does it look like to quickly flash an "unknown" before reseting it, but that might be weird too. I'm not so so worried about this as we can improve it in the future

Reply:
easily done.. changing now